middleware/kubernetes/parse.go

```diff
@@ -101,6 +98,5 @@ func (r recordRequest) String() string {
 	s += "." + r.service
 	s += "." + r.namespace
 	s += "." + r.podOrSvc
-	s += "." + r.zone
```

Comment:
strings.Join requires a string slice to be constructed. If you worried about mem...